src/ethereum/constantinople/vm/interpreter.py

```diff
@@ -164,6 +166,51 @@ def process_create_message(message: Message, env: Environment) -> Evm:
     return evm
 
 
+def process_create2_message(message: Message, env: Environment) -> Evm:
+    """
+    Executes a call to create a smart contract via CREATE2 opcode.
+
+    Parameters
+    ----------
+    message :
+        Transaction specific items.
+    env :
+        External items required for EVM execution.
+
+    Returns
+    -------
+    evm: :py:class:`~ethereum.constantinople.vm.Evm`
+        Items containing execution specific objects.
+    """
+    # take snapshot of state before processing the message
+    begin_transaction(env.state)
+
+    # It's expected that the creation operation works on empty storage. Hence
+    # we delete the storage and restore the account's state if there is an
+    # error in the initialization code execution.
+    destroy_storage(env.state, message.current_target)
```

Comment:
For anyone reviewing this: The only difference between this method and `process_create_message` is this line. I was wondering if we could re-use `process_create_message` but I can't think of a way that would be specs appropriate.